### PR TITLE
move `isDestroyed` check from `getPath` to `get`

### PR DESCRIPTION
**Pull request by bekzod**
make consistent `get` with `set` so it warns (maybe make it `assert` later) when used on destroyed object, I think it will help to discover hidden bugs

### CODE REVIEW

File: packages/ember-metal/lib/is_empty.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import { get } from './property_get';
```
Comment (rwjblue) [inline packages/ember-metal/lib/is_empty.ts]: This is a separate, unrelated change. Can you keep this PR focused specifically on changing `get` and `getPath` and send another PR for these changes?

File: packages/ember-metal/lib/property_get.ts [hunk 2]
```diff
@@ -97,6 +94,16 @@ export function get(obj: object, keyName: string): any {
...
   assert('Cannot call `get` with an empty string', keyName !== '');
+  if (
+    (obj as ExtendedObject).isDestroyed &&
+    keyName !== 'isDestroyed' &&
+    keyName !== 'isDestroying'
+  ) {
+    warn(`calling \`get\` on destroyed object: ${toString(obj)}.${keyName}`, false, {
+      id: 'calling-get-on-destroyed',
+    });
+    return;
```
Comment (rwjblue) [inline packages/ember-metal/lib/property_get.ts]: In general, I think a warning when accessing properties on destroyed objects is probably fine, but I do not think we should prevent accessing the underlying state (at least not right away).

Can you refactor this so that the conditionals are all part of the `warn` call (to ensure that it is all removed properly from production builds) and remove the early return?

File: packages/ember-metal/lib/property_get.ts [hunk 3]
```diff
@@ -97,6 +94,16 @@ export function get(obj: object, keyName: string): any {
     typeof keyName !== 'string' || keyName.lastIndexOf('this.', 0) !== 0
   );
   assert('Cannot call `get` with an empty string', keyName !== '');
+  if (
+    (obj as ExtendedObject).isDestroyed &&
```
Comment (rwjblue) [inline packages/ember-metal/lib/property_get.ts]: We should modify these `isDestroyed` checks to be using `meta.sourceIsDestroyed` instead...

File: packages/ember-views/lib/system/event_dispatcher.js [hunk 4]
```diff
@@ -360,6 +360,10 @@ export default EmberObject.extend({
   },
 
   destroy() {
+    if (this.isDestroyed) {
```
Comment (rwjblue) [inline packages/ember-views/lib/system/event_dispatcher.js]: Are we calling `destroy` multiple times?

**Review by rwjblue — CHANGES_REQUESTED** [review]
I'm not certain here.  I

File: packages/ember-views/lib/system/event_dispatcher.js (hunk 4 above)
Comment (bekzod) [reply packages/ember-views/lib/system/event_dispatcher.js]: yep, was thing of adding warning to `destroy` for calling multiple times what do you think ?

File: packages/ember-metal/lib/property_get.ts (hunk 3 above)
Comment (bekzod) [reply packages/ember-metal/lib/property_get.ts]: that is how it was done before and same way done in `property_set`, maybe make it later with another PR just to keep things incremental so easier to track things if there is issue ?

File: packages/ember-metal/lib/property_get.ts (hunk 2 above)
Comment (bekzod) [reply packages/ember-metal/lib/property_get.ts]: so I should also remove `return` so it doesn't prevent accessing property of destroyed object ?

File: packages/ember-metal/lib/is_empty.ts (hunk 1 above)
Comment (bekzod) [reply packages/ember-metal/lib/is_empty.ts]: it is a bit related :P , I did so to avoid warning for calling `isEmpty` on destroyed object in tests

File: packages/ember-metal/lib/property_get.ts (hunk 2 above)
Comment (rwjblue) [reply packages/ember-metal/lib/property_get.ts]: yes, and remove the conditional (moving those conditions to the `warn` itself)

File: packages/ember-metal/lib/property_get.ts (hunk 3 above)
Comment (rwjblue) [reply packages/ember-metal/lib/property_get.ts]: Yep, totally agreed! I was just trying to point out the issue generally, it definitely isn't related to this PR at all. I'm sorry I didn't mention it in this comment.

File: packages/ember-metal/lib/property_get.ts (hunk 2 above)
Comment (bekzod) [reply packages/ember-metal/lib/property_get.ts]: done, but `warn` seems to not lazily evaluate string part of warn

Comment (bekzod) [reply packages/ember-metal/lib/property_get.ts]: added PR to `babel-plugin-debug-macros` https://github.com/chadhietala/babel-plugin-debug-macros/pull/72 to support lazy evaluation of string part of warn

File: packages/@ember/-internals/metal/lib/property_get.ts [hunk 5]
```diff
@@ -90,6 +92,20 @@ export function get(obj: object, keyName: string): any {
     typeof keyName !== 'string' || keyName.lastIndexOf('this.', 0) !== 0
   );
 
+  if (DEBUG && !TOLERATE_DESTROYED) {
+    let objString: string;
+    TOLERATE_DESTROYED = true;
```
Comment (bekzod) [inline packages/@ember/-internals/metal/lib/property_get.ts]: added `TOLERATE_DESTROYED` flag, to avoid getting warning when `get` called within `toString`

**Comment by bekzod** [comment]
@rwjblue minding giving another peek

**Comment by bekzod** [comment]
oops test failing
